pandas/core/frame.py

```diff
@@ -3733,6 +3735,36 @@ def _set_item(self, key, value) -> None:
                 if isinstance(existing_piece, DataFrame):
                     value = np.tile(value, (len(existing_piece.columns), 1)).T
 
+        if (
```

Comment:
we don't want to check anything here (in the frame), rather in indexing.py as a lot of things happen during setitem, you are short-circuiting and likely causing a number of tests to fail.

start really simple. make small incremental changes.

Reply:
Thanks. I've made some progress here #41193 but I am still having some issues. Could you have a look there instead? @jreback